File: jobs/prueba/feecalc.py
from __future__ import annotations

from dataclasses import dataclass
from decimal import Decimal


@dataclass(frozen=True)
class FeeResult:
    balance: Decimal
    min_balance: Decimal
    fee: Decimal
    new_balance: Decimal
    fee_applied: bool


def calculate_fee(
    balance: Decimal,
    min_balance: Decimal = Decimal("500.00"),
    fee: Decimal = Decimal("12.50"),
) -> FeeResult:
    if balance < min_balance:
        new_balance = balance - fee
        return FeeResult(balance, min_balance, fee, new_balance, True)
    return FeeResult(balance, min_balance, fee, balance, False)


def format_currency(amount: Decimal) -> str:
    return f"${amount:,.2f}"


def main() -> FeeResult:
    result = calculate_fee(Decimal("450.75"))
    print("FEECALC START")
    print(f"BALANCE={result.balance}")
    print(f"MIN={result.min_balance}")
    if result.fee_applied:
        print(f"FEE APPLIED={result.fee}")
        print(f"NEW BALANCE={format_currency(result.new_balance)}")
    else:
        print("NO FEE")
    print("FEECALC END")
    return result


if __name__ == "__main__":
    main()

File: jobs/f355bebc-0b74-4e1d-b82c-c780265b309d/feecalc.java
/** Monthly maintenance fee on low-balance accounts (modernized from FEECALC.cbl). */

import java.math.BigDecimal;
import java.math.RoundingMode;
import java.text.NumberFormat;
import java.util.Locale;

public final class feecalc {

    private feecalc() {}

    public record FeeResult(
            BigDecimal balance,
            BigDecimal minBalance,
            BigDecimal fee,
            BigDecimal newBalance,
            boolean feeApplied) {}

    private static BigDecimal quantize(BigDecimal value) {
        return value.setScale(2, RoundingMode.HALF_UP);
    }

    public static FeeResult calculateFee(
            BigDecimal balance, BigDecimal minBalance, BigDecimal fee) {
        if (balance.compareTo(minBalance) < 0) {
            BigDecimal newBalance = quantize(balance.subtract(fee));
            return new FeeResult(balance, minBalance, fee, newBalance, true);
        }
        return new FeeResult(balance, minBalance, fee, balance, false);
    }

    public static String formatCurrency(BigDecimal amount) {
        return NumberFormat.getCurrencyInstance(Locale.US).format(amount);
    }

    public static void main(String[] args) {
        FeeResult result =
                calculateFee(
                        new BigDecimal("450.75"),
                        new BigDecimal("500.00"),
                        new BigDecimal("12.50"));

        System.out.println("FEECALC START");
        System.out.println("BALANCE=" + result.balance());
        System.out.println("MIN=" + result.minBalance());
        if (result.feeApplied()) {
            System.out.println("FEE APPLIED=" + result.fee());
            System.out.println("NEW BALANCE=" + formatCurrency(result.newBalance()));
        } else {
            System.out.println("NO FEE");
        }
        System.out.println("FEECALC END");
    }
}
